### DISCUSSION TITLE
[Feature] Claim obsolete package names

### DISCUSSION THREAD

**Discussion by codemaster138** [npm/rfcs#361] [RRFC]
It happens almost every time: I think of a great name for my project, I look it up on npm to check if the name is available, and what do I find? A package with `readme` that says "WIP", no github repo, that hasn't been updated in 3 Years.

As the registry's namespace becomes increasingly saturated, this problem is becoming worse and worse. I want to propose a solution:

A registry that automatically assigns the tag "obsolete" to packages that haven't been updated for more than 2 Years, and that have had less than 500 downloads in the last year (Numbers are examples, better ones can be worked out later if this idea hits support). Anyone who wants to use this package name can then run `npm claim` to claim it.

I am not sure what the best claiming mechanism is yet. I was thinking that maybe the claim gets sent to the package owner, who is *loudly* notified over email and a notification from the NPM frontend. If the old package owner fails to deny the claim within 2 months, the package claim goes through. The previous owner can also proactively accept the claim to make it go through immediately.

While waiting for the claim to go through, the package goes into a "trial" period. In this period, the claiming party can publish a package to the package name, but the content they publish is only available to said party, the rest of the world would still see the old package.

If the claimee denies the claim, the trial pushes of the claiming party are unpublished.

I'd like to know what the community thinks about this idea, especially since I don't have the full picture of what this feature would like yet. I personally think this would be a good way to free up some namespace since, with 1,577,063 packages on NPM, it's really getting kinda tight...

**Comment by ljharb** [comment]
Plenty of packages that haven’t been touched in many years are still downloaded and used. Your suggestion would break lots of people for minimal value, since you can use a scoped package and avoid any conflicts.

That said, there are many valid use cases for claiming a package name, and typically this is done via npm support.

**Reply by tjor75** [reply]
It would be useful for test packages that only contain a `console.log` , though
